doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -493,40 +495,257 @@ If an error is encountered during the `rollback` operation, an error is reported
 
+### 2.2.4 Replace
+The SONiC `replace` command can broadly classified into the following steps
+
+#### Stage-1 Get current ConfigDB JSON config
+The *ConfigDBConnector* class is used to obtain the running configuration in JSON format
+
+#### Stage-2 Get target config from the external user/system
+The external user/system to provide the full target ConfigDB config in JSON format.
+
+#### Stage-3 Validating the target config using YANG models
+The target config are unknown configs and need to be validating using YANG models
+
+#### Stage-4 Generate the diff as JsonPatch between current config and target config
+The current ConfigDB JSON config is compared with the target JSON config. The comparison result should be in JsonPatch format.
+
+#### Stage-5 Apply-Patch
+Pass the generated JsonPatch to the apply-patch API
+
+#### Stage-6 Verify config replace
+Compare the ConfigDB JSON after the update with the target JSON, there should be no differences.
+
+#### Fail-safe Action
+
+If an error is encountered during the `replace` operation, an error is reported and the system DOES NOT take any automatic action. The user can take a `checkpoint` before running `replace` and if the operation failed, the user can `rollback`.
+
+#### Logging
+
+All the configuration update operations executed and the output displayed by the `replace` command are stored in the systemd journal. They are also forwarded to the syslog. By storing the commands in the systemd-journal, the user will be able to search and display them easily at a later point in time. The `show replace log` command reads the systemd-journal to display information about the `replace` command that was previously executed or currently in progress.
+
 # 3 Design
 
 ## 3.1 Overview
 
 ### 3.1.1 ApplyPatch
-![apply-patch-design](files/apply-patch-design.png)
+<img src="files/apply-patch-design.png" alt="apply-patch-design" width="1200"/>
+
+
+#### 3.1.1.1 User
+The user of the system, can simply be a human operator or a service that can talk to SONiC CLI. The user can only apply-patch if they have an admin permission. Users without admin permissions can only execute dry-run of the operations where they will be able to see the exact changes going to affect the device, without executing these changes.
+
+#### 3.1.1.2 SONiC CLI
+These are the CLI of SONiC switch to which makes it easy for the users to interact with the system. The CLI commands we are interested in are `config ...` and `show ...`, check [SONiC Command Line Interface Guide](https://github.com/Azure/sonic-utilities/blob/master/doc/Command-Reference.md) to learn more about SONiC CLI.
```

Comment:
Here the user will see only the config changes after dry-run or they will also see the list of services that will be restarted due to the config changes.

Reply:
The plan is to see both.

We will execute the whole code as expected, and will replace the last step when actually updating configdb with a specific config or issuing a service restart/service validation commands with printing to user.